lib/ws_servers/api/send_authenticated.js

```diff
@@ -55,6 +55,6 @@ module.exports = async (ws, db, marketData, d, opts) => {
   const { UserSettings } = db
   const { userSettings } = await UserSettings.getAll()
 
-  ws.algoWorker.start({ apiKey: key, apiSecret: secret, userId: DEFAULT_USER })
+  await ws.algoWorker.start({ apiKey: key, apiSecret: secret, userId: DEFAULT_USER })
```

Comment:
If this fails, the host won't be available .. so, may be we should add a check in algo worker's methods to see if host is connected .. or check if the `algoworker.isStarted` is true before calling algoworker methods ? Otherwise, we will get a lot of errors `Cannot read property {method} of null`.

Reply:
I'll map which methods are being used publicly and  add the checks

Reply:
done